### ISSUE TITLE
Error: Could not open file kaiju.out

### ISSUE THREAD

**Issue by ZoeHansen** [bioinformatics-centre/kaiju#169]
Hello! Thank you for curating Kaiju, and for monitoring this GitHub so closely! I am trying to run kaiju and am being met with this error: 

`Error: Could not open file ./kaiju/output/ER0043.kaiju.out
`

My script looks like this:

`./kaiju/bin/kaiju -t ./kaiju/kaijudb/nodes.dmp -f ./kaiju/kaijudb/kaiju_db_nr_euk.fmi -i $HOME/amrplusplus/test/$i.non.host.R1.fastq.gz -j $HOME/amrplusplus/test/$i.non.host.R2.fastq.gz -o ./kaiju/output/$i.kaiju.out;`

I am running kaiju in a loop and have designated my output file as $i.kaiju.out to allow my sample IDs to be the prefix of the file, but am continuing to get this error. It seems as though kaiju is not recognizing my -o setting correctly. Am I constructing the syntax incorrectly? Does -o need to be designated in some other way? 

Thank you for your help!

**Comment by pmenzel** [comment]
Does the folder `./kaiju/output` already exist?

**Comment by ZoeHansen** [comment]
Hi again -- yes, I had made an 'output' directory to hold these files. Is kaiju trying to write this directory rather than navigating into it? 

Thank you for your time!

**Comment by pmenzel** [comment]
Maybe it's a permission problem? Try to create a file in the output folder
manually. If that works, then kaiju should also be able to create a file
there.

**Comment by ZoeHansen** [comment]
I am able to write a file there. In fact, my kaiju2table command works just fine and outputs an empty 'sample.summary.tsv' into my output directory, it's just the kaiju.out file that is not being written with the basic 'kaiju' command. 

I can try to keep playing around with it, but do you know if something/what could be going wrong? Thank you, again!

**Comment by ZoeHansen** [comment]
Hello again! Alright, I've been trying to figure out what's been going on with my kaiju.out files, but when I attempted to run my script again my 'kaijudb' directory which houses my database was empty (for reasons unknown, haha). I had originally tried to use the 'kaiju-makedb' command to download the nr_euk database, however, I continue to get this error when running this command: 

`./bin/kaiju-makedb: line 224: 43539 Bus error               (core dumped) kaiju-mkbwt -e $exponentSA_NR -n $threadsBWT -a ACDEFGHIKLMNPQRSTVWY -o $DB/kaiju_db_$DB $DB/kaiju_db_$DB.faa
`

This error appears ~6 hours after the database download has begun. 

Earlier, I just tried using 'wget' and using the database from the webserver, but this is the version that disappeared somehow. I'd like to use 'kaiju-makedb' if possible (unless the webserver database disappearance was a fluke)-- could this also be what's playing into my problems with the 'kaiju.out' files?? 

I'm sorry for the convoluted problems -- thank you again for your help!

**Comment by pmenzel** [comment]
a crash in kaiju-mkbwt usually means that you don't have enough RAM on the server to make the database. see the README.txt for the memory requirements (which are already a bit outdated, so you likely need more)

**Comment by ZoeHansen** [comment]
@pmenzel Okay, thank you! I've tried running kaiju-makedb with upwards of 275GB which is ~80GB more than the README calls for and it is still failing -- does it need more than this? I'll keep trying trial-and-error, but if you have a specific RAM suggestion, I'd really appreciate it. Thanks!

**Comment by pmenzel** [comment]
I run `kaiju-makedb -s nr_euk` today and it took ~12 hours and 224 GB of RAM.

**Comment by ZoeHansen** [comment]
@pmenzel Thank you for the information! I had been able to download the database using 350GB of RAM and this took approximately 6-7 hours (I decided to drastically overestimate for RAM to see if it would work, and it did!). 

I was still having trouble with my initial error ('could not open file kaiju.out') and so I tried writing an empty file prior to running kaiju and this seemed to work. It seems that kaiju couldn't create the output file, but could write to an existing file. I'm sorry if I misunderstood your documentation, but everything seems to be working just fine now!

Thank you for all of your help and for responding so promptly!
